.github/workflows/pre-commit.yaml

```diff
@@ -19,4 +23,6 @@ jobs:
           gradle-version: 8.7
 
       - name: Build with Gradle
+        env:
+          BOT_TOKEN: ${{ secrets.BOT_TOKEN }}
```

Comment:
We could rename this secret variable to something like `DISCORD_BOT_TOKEN` or `TEST_BOT_TOKEN` to increase clarity and to indicate that it is to be used by the integration tests.

Reply:
Thanks for reviewing, I'm going to skip this one change just so we can get the testing in place merged. Since we're going to continually be doing work on the testing and many more PRs will come.